core/server/src/main/java/alluxio/master/file/FileSystemMaster.java

```diff
@@ -2814,17 +2814,41 @@ public FileSystemCommand workerHeartbeat(long workerId, List<Long> persistedFile
       inode.setLastModificationTimeMs(opTimeMs);
     }
     if (options.getTtl() != null) {
-      Preconditions.checkArgument(inode.isFile(), PreconditionMessage.TTL_ONLY_FOR_FILE);
       long ttl = options.getTtl();
-      InodeFile file = (InodeFile) inode;
-      if (file.getTtl() != ttl) {
-        mTtlBuckets.remove(file);
-        file.setTtl(ttl);
-        mTtlBuckets.insert(file);
-        file.setLastModificationTimeMs(opTimeMs);
-        file.setTtlAction(options.getTtlAction());
+      if (inode.isDirectory()) {
+        InodeDirectory directory = (InodeDirectory) inode;
+        InodeDirectory parent = inodePath.getParentInodeDirectory();
+        if (parent != null) {
+          System.out.println("parent:" + parent.getTtl() + " " + ttl);
```

Comment:
Please use `LOG.error()` instead of `System.out.println` which will not be actually printed